compute/volumes.go

```diff
@@ -0,0 +1,216 @@
+//
+// Copyright (c) 2018, Joyent, Inc. All rights reserved.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+
+package compute
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"path"
+
+	"github.com/joyent/triton-go/client"
+	"github.com/pkg/errors"
+)
+
+type VolumesClient struct {
+	client *client.Client
+}
+
+type Volume struct {
+	ID             string   `json:"id"`
+	Name           string   `json:"name"`
+	Owner          string   `json:"owner_uuid"`
+	Type           string   `json:"type"`
+	FileSystemPath string   `json:"filesystem_path"`
+	State          string   `json:"state"`
+	Networks       []string `json:"networks"`
+	Refs           []string `json:"refs"`
+}
+
+type ListVolumesInput struct {
+	Name  string
+	Size  string
+	State string
+	Type  string
+}
+
+func (c *VolumesClient) List(ctx context.Context, input *ListVolumesInput) ([]*Volume, error) {
+	fullPath := path.Join("/", c.client.AccountName, "volumes")
+
+	query := &url.Values{}
+	if input.Name != "" {
+		query.Set("name", input.Name)
```

Comment:
This isn't critical for this PR, but `"name"`, `"size"`, `"state"`, and `"type"` should all be moved into a `const` block in the package so that we can find all call sites where a given parameter is used.  I'd probably use an unexported `const` identifier, too.  Something like:

```go
const (
  _VolumeName = "name"
)
```

If this were internal only and the inputs weren't coming from the caller, I'd suggest something like:

```go
type VolumeAttr string

const (
  _VolumeName _VolumeAttr = "name"
)
```

The only reason I'm saying that is `"name"` is a string with no type and isn't bound specifically to the Volume API and is a generic value that could be reused or hard to track down in the future.  Not critical, but something I've found that makes navigating code a bit easier, albeit more tedious to write up front.  Because this is at a customer-facing boundary, I'd only do the former, not the latter.

The extra commentary is only here for future code anthropologists.